src/mlpack/core/optimizers/sgd/test_function.hpp

```diff
@@ -24,10 +24,16 @@ namespace test {
 //! iterations.
 class SGDTestFunction
 {
+ private:
+  arma::Col<size_t> visitationOrder;
+
  public:
   //! Nothing to do for the constructor.
   SGDTestFunction() { }
 
+  //! Shuffling the elements
```

Comment:
We should be more clear about the documentation here.  I would suggest something more like

```
/**
 * Shuffle the order of function visitation.  This may be called by the optimizer.
 */
```